swarm/network/stream/common_test.go

```diff
@@ -317,3 +320,30 @@ func createTestLocalStorageForID(id enode.ID, addr *network.BzzAddr) (storage.Ch
 	}
 	return store, datadir, nil
 }
+
+// watchDisconnections receives simulation peer events in a new goroutine and sets atomic value
+// disconnected to true in case of a disconnect event.
+func watchDisconnections(ctx context.Context, sim *simulation.Simulation) (disconnected atomic.Value) {
+	log.Debug("Watching for disconnections")
+	disconnections := sim.PeerEvents(
+		context.Background(),
```

Comment:
Do we intentionally create a new context and not pass `ctx`? If yes, can you tell why?

Reply:
That was in the original code in this test. It actually should pass the context.